sonar-db/src/main/java/org/sonar/db/component/SnapshotDao.java

```diff
@@ -42,11 +43,19 @@ public static boolean isLast(SnapshotDto snapshotTested, @Nullable SnapshotDto p
     return previousLastSnapshot == null || previousLastSnapshot.getCreatedAt() < snapshotTested.getCreatedAt();
   }
 
+  /**
+   * @deprecated use {@link #selectByUuid(DbSession, String)}
+   */
+  @Deprecated
   @CheckForNull
   public SnapshotDto selectById(DbSession session, long id) {
```

Comment:
I've checked, it seems we will be able to drop `selectById` and `selectOrFailById` when we remove the use of SNAPSHOTS.PARENT_ID and SNAPSHOTS.ROOT_ID